File: LeetCode/Medium/0179-largest-number/0179-largest-number.py
class Solution:
    def largestNumber(self, nums: List[int]) -> str:
        nums = [str(v) for v in nums]

        def compare(a, b):
            if a + b > b + a:
                return -1
            return 1

        nums.sort(key=cmp_to_key(compare))

        return "".join(nums).lstrip("0") or "0"

File: LeetCode/Medium/0179-largest-number/0179-largest-number.go
func largestNumber(nums []int) string {
    numStrs := make([]string, len(nums))
    
	for i, num := range nums {
		numStrs[i] = strconv.Itoa(num)
	}

	sort.Slice(numStrs, func(i, j int) bool {
		return numStrs[i]+numStrs[j] > numStrs[j]+numStrs[i]
	})

	if numStrs[0] == "0" {
		return "0"
	}

	result := ""
	for _, num := range numStrs {
		result += num
	}

	return result
}